### PR TITLE
added simple ping for client/server

### LINKED ISSUE #38: Existing port binding is not always respected when choosing a local port.
How to reproduce:
On Mac, start a docker image that is bound to port 8081, e.g. `docker run -p 8081:5000 training/webapp python app.py` (I used Docker version 1.12.3).
Then run `python3 -m zign` and it will use port `8081` but Chrome (and Safari) resolves `http://localhost:8081` to the docker image.

It seems to be related to ipv4 and ipv6 since `http://[::1]:8081` goes to the docker image while `http://127.0.0.1:8081` goes to server spawned by zign.

### PR DESCRIPTION
**Pull request by olevchyk**
simple  HELO/EHLO request on ClientRedirectServer to check its availability, completely fixes https://github.com/zalando-stups/zign/issues/38

### CODE REVIEW

File: zign/api.py [hunk 1]
```diff
@@ -43,6 +44,13 @@ def __str__(self):
         return 'Configuration error: {}'.format(self.msg)
 
 
+def open_browser(url=None):
+    user = os.getenv('USER') 
+    if os.getegid() == 0 and user and user != "root":
```
Comment (hjacobs) [inline zign/api.py]: This looks like a hack, why would we want to support that?

File: zign/api.py [hunk 2]
```diff
@@ -18,7 +19,7 @@
 from .oauth2 import ClientRedirectServer
 
 TOKEN_MINIMUM_VALIDITY_SECONDS = 60*5  # 5 minutes
-
+REDIRECT_SERVER_HOST = "127.0.0.1"
```
Comment (hjacobs) [inline zign/api.py:21]: Changing the `redirect_uri` only works when the IAM/OAuth provider also knows about this changed `redirect_uri` (no way we can roll it out at once, due to the distributed nature of CLI tools).

Comment (olevchyk) [reply zign/api.py:21]: ah I see that makes total sense, removed this from being used for redirection, left only server part.

File: zign/api.py (hunk 1 above)
Comment (olevchyk) [reply zign/api.py]: just a proposal, removed this part from PR.

**Pull request closed without merging**